python/src/iceberg/table/schema.py

```diff
@@ -0,0 +1,286 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Dict, Generic, Iterable, List, TypeVar
+
+from iceberg.types import (
+    IcebergType,
+    ListType,
+    MapType,
+    NestedField,
+    PrimitiveType,
+    StructType,
+)
+
+T = TypeVar("T")
+
+
+class Schema:
+    """A table Schema"""
+
+    def __init__(self, *columns: Iterable[NestedField], schema_id: int, identifier_field_ids: List[int] = []):
+        self._struct = StructType(*columns)  # type: ignore
+        self._schema_id = schema_id
+        self._identifier_field_ids = identifier_field_ids
+        self._name_index: Dict[str, int] = index_by_name(self)
+        self._id_index: Dict[int, NestedField] = {}  # Will be lazily set when self.id_index property method is called
+
+    def __str__(self):
+        return "table { \n" + "\n".join([" " + str(field) for field in self.columns]) + "\n }"
+
+    def __repr__(self):
+        return f"Schema(fields={repr(self.columns)})"
+
+    @property
+    def columns(self) -> Iterable[NestedField]:
+        return self._struct.fields
+
+    @property
+    def id(self) -> int:
```

Comment:
I think this should be `schema_id` to be clear.

Reply:
Updated